label_studio/storage/redis.py

```diff
@@ -0,0 +1,154 @@
+import json
+import os
+import logging
+import redis
+
+from .base import BaseStorage, BaseForm, StringField
+
+
+logger = logging.getLogger(__name__)
+
+def get_redis_connection(db = None):
+    if not db:
+        # This should never happen, but better to check than to accidentally 
+        # overwrite an existing database by choosing a wrong default:
+        logger.error("No redis db id passed!")
+    # Since tasks are always text, we use StrictRedis with utf-8 decoding.
+    return redis.StrictRedis(db=db, charset="utf-8", decode_responses=True)
+
+
+class RedisStorageForm(BaseForm):
+    path = StringField('Path', description='Storage prefix (optional)')
+    bound_params = dict(path='path')
+
+
+class RedisStorage(BaseStorage):
+    """Storage class for a redis database connection. 
+    
+    Inherits from BaseStorage.
+
+    Conventions:
+    A redis server hosts multiple, enumerated databases (0,1,2...).
+    The database used for storage should ONLY contain tasks. Any data already
+    in there might be lost / overwritten at any point in time and / or break
+    the functionality.
+
+    All keys are integers, but redis saves them as strings, so we sometimes
+    need to convert. The redis key is our prefix (e.g. "my_project/") + the 
+    labelstudio key as string (e.g. "3").
+
+    Example: If path is defined as "my_project", calling set(1, my_val) will
+    write to 
+    my_project/1 (the "/" is added if not already in path)
+    
+    Values are python dicts encoded as utf-8 strings.
+    """
+
+    description = 'Redis database connection'
+    
+    form = RedisStorageForm
+
+    def __init__(self, project_path, path=None, db=1, **kwargs):
```

Comment:
do we have some roadblocks for adding other crucial connection parameters at once, like host, port, and password? that would allow using more general Redis connections besides the default localhost

Reply:
This was supposed to end up in this conversation:

No, should be an easy task, I will add a few more after having a look at
what additional parameters might be useful.

Reply:
I will instead try to implement a way of forwarding **kwargs since there are so many parameters that also might change in the future, and have the GUI support a subset.

Reply:
Adding generic parameters proved difficult since I don't fully understand what is happening in the front-end. I instead added host, port and password as parameters to the form GUI & backend in the updated PR.

> I will instead try to implement a way of forwarding **kwargs since there are so many parameters that also might change in the future, and have the GUI support a subset.